MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -939,6 +939,7 @@
 				((_vkFormat == VK_FORMAT_R32_UINT || _vkFormat == VK_FORMAT_R32_SINT) ||
 					(_hasMutableFormat && pixFmts->getViewClass(_vkFormat) == MVKMTLViewClass::Color32 && (getIsValidViewFormat(VK_FORMAT_R32_UINT) || getIsValidViewFormat(VK_FORMAT_R32_SINT))));
 
+	_isLinearForAtomics = false;
 	if (_shouldSupportAtomics && !getPhysicalDevice()->useNativeTextureAtomics())
 		_isLinearForAtomics = _arrayLayers == 1 && getImageType() == VK_IMAGE_TYPE_2D;
```

Comment:
Probably better to combine these into one operation to improve stability and readability.

```suggestion
	_isLinearForAtomics = _shouldSupportAtomics && !getPhysicalDevice()->useNativeTextureAtomics() && _arrayLayers == 1 && getImageType() == VK_IMAGE_TYPE_2D;
```